core/thread/id.go

```diff
@@ -184,84 +171,138 @@ func uvError(read int) error {
 	}
 }
 
+func (i ID) Valid() bool {
+	data := i.Bytes()
+	err := validateIDData(data)
+	return err == nil
+}
+
+func getVersion(data []byte) (uint64, int, error) {
+	vers, n := binary.Uvarint(data)
+	if err := uvError(n); err != nil {
+		return 0, 0, err
+	}
+	return vers, n, nil
+}
+
+func validateIDData(data []byte) error {
+	vers, n, err := getVersion(data)
+	if err != nil {
+		return err
+	}
+
+	if vers != V1 {
+		return fmt.Errorf("expected 1 as the id version number, got: %d", vers)
+	}
```

Comment:
First we get the version then make sure it is a known version (currently just `V1`)